### PR TITLE
Respect server-side toggle for platform checkout eligibility

### LINKED ISSUE #1730: Implement "Digital wallets" settings screen UI
## Description

Blocked by https://github.com/Automattic/woocommerce-payments/issues/1508
See paJDYF-1x5-p2 for latest updates on the screens UI.
Saving of settings will be done in a different task: https://github.com/Automattic/woocommerce-payments/issues/1728

## Acceptance criteria
- The page shows the UI elements, which can be interacted with.

### LINKED ISSUE #3956: Respect server-side toggle for platform checkout eligibility
### Description

Retrieve server-side platform checkout eligibility flag. If the merchant is eligible, we should enable merchants to turn on the the platform checkout setting.

### Acceptance criteria

* Replace feature flag with server-side toggle.

### Additional context

pdpOw2-sg-p2

### PR DESCRIPTION
**Pull request by brianyu28**
Fixes #3956

#### Changes proposed in this Pull Request

Currently, whether platform checkout can be enabled is determined by an option (`_wcpay_feature_platform_checkout`) set on the merchant site. 1731-gh-Automattic/woocommerce-payments-server, we introduce a server-side flag to control platform checkout eligibility instead. This PR replaces WCPay's local feature flag check with the server-side flag to determine platform checkout eligibility.

#### Testing instructions

This PR depends on and should be tested alongside 1731-gh-Automattic/woocommerce-payments-server in WCPay Server.

- In your WCPay Server database, check the `wcpay_accounts` table. Find the account that corresponds to your merchant site, and take note of its `wcpay_account_id`.
- In your WCPay Server database, go to the `wcpay_account_meta` table and add a row for the account ID you took note of in the previous step. Set the `platform_checkout_eligible` meta key to `0`.
- In your WCPay merchant site, go to WCPay Dev Tools and "clear" your "Account cache contents."
- Look at your "Account cache contents" and verify that the `account`'s `platform_checkout_eligible` field is set to `false`.
- In WCPay settings, note that there's no checkbox to enable platform checkout. On the checkout page, platform checkout should not be available.
- Go back to your `wcpay_account_meta` table and set `platform_checkout_eligible` to `1`.
- In your WCPay merchant site, go to WCPay Dev Tools and "clear" your "Account cache contents."
- Look at your "Account cache contents" and verify that the `account`'s `platform_checkout_eligible` field is set to `true`.
- In WCPay settings, note that there's a checkbox to enable platform checkout. Verify that platform checkout features work only when that checkbox is enabled _and_ `platform_checkout_eligible` is `true`.

Note that, in the WCPay client code, this PR makes the distinction between `platform_checkout_eligible` (the merchant is _allowed_ to enable platform checkout) and `platform_checkout_enabled` (the merchant has _turned on_ platform checkout).

Additional Context: pdpOw2-sg-p2

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) : _Add link here / 'QA Testing Not Applicable'_

### CODE REVIEW

**Review by alefesouza — APPROVED** [review]
Great work! LGTM, let's work await @leonardola review

File: includes/class-wc-payments-features.php [hunk 1]
```diff
@@ -13,8 +13,8 @@
  * WC Payments Features class
  */
 class WC_Payments_Features {
-	const UPE_FLAG_NAME = '_wcpay_feature_upe';
-
+	const PLATFORM_CHECKOUT_FLAG_NAME   = 'platform_checkout_eligible';
```
Comment (leonardola) [inline includes/class-wc-payments-features.php:16]: What do think about changing the `PLATFORM_CHECKOUT_FLAG_NAME` to `PLATFORM_CHECKOUT_ELIGIBLE_FLAG_NAME` ? 	The name used could be misleading

File: tests/unit/test-class-wc-payments-features.php [hunk 2]
```diff
@@ -72,33 +73,18 @@ function ( $pre_option, $option, $default ) {
...
-	 * @dataProvider is_platform_checkout_falsy_value_provider
-	 */
 	public function test_is_platform_checkout_is_returned_as_false_if_not_equal_1() {
-		update_option( '_wcpay_feature_platform_checkout', '0' );
-		$this->assertFalse( WC_Payments_Features::is_platform_checkout_enabled() );
+		add_option( WC_Payments_Account::ACCOUNT_OPTION, [ 'account' => [ 'platform_checkout_eligible' => false ] ] );
+		$this->assertFalse( WC_Payments_Features::is_platform_checkout_eligible() );
 	}
 
 	public function test_is_platform_checkout_is_returned_as_false_if_missing() {
```
Comment (leonardola) [inline tests/unit/test-class-wc-payments-features.php:85]: This kinda looks misspelled `test_is_platform_checkout_is_returned_as_false_if_missing` maybe it was meant to be `test_if`... or maybe `test_is_platform_checkout_ineligible_by_default`

File: tests/unit/test-class-wc-payments-features.php [hunk 3]
```diff
@@ -72,33 +73,18 @@ function ( $pre_option, $option, $default ) {
 	}
 
 	public function test_is_platform_checkout_is_returned_as_true() {
```
Comment (leonardola) [inline tests/unit/test-class-wc-payments-features.php:75]: This `test_is_platform_checkout_is_returned_as_true` kinda looks misspelled. Maybe it was meant to be `test_if`... or even `test_is_platform_checkout_eligible`

File: tests/unit/test-class-wc-payments-features.php [hunk 4]
```diff
@@ -72,33 +73,18 @@ function ( $pre_option, $option, $default ) {
...
-		update_option( '_wcpay_feature_platform_checkout', '1' );
-		$this->assertTrue( WC_Payments_Features::is_platform_checkout_enabled() );
+		add_option( WC_Payments_Account::ACCOUNT_OPTION, [ 'account' => [ 'platform_checkout_eligible' => true ] ] );
+		$this->assertTrue( WC_Payments_Features::is_platform_checkout_eligible() );
 	}
 
-	/**
-	 * @dataProvider is_platform_checkout_falsy_value_provider
-	 */
 	public function test_is_platform_checkout_is_returned_as_false_if_not_equal_1() {
```
Comment (leonardola) [inline tests/unit/test-class-wc-payments-features.php:80]: This `test_is_platform_checkout_is_returned_as_false_if_not_equal_1` kinda looks misspelled. Maybe it was meant to be `test_if`... or even `test_is_platform_checkout_ineligible_if_not_equal_1`

**Review by leonardola — APPROVED** [review]
Nice work on this one. 
I've tested it and it works as expected. Left some nitpicks about some test names and const name. But If you disagree feel free to merge as is.

File: includes/class-wc-payments-features.php (hunk 1 above)
Comment (brianyu28) [reply includes/class-wc-payments-features.php]: Makes sense to me, updated in 704ab29f5533d1673e631ad089ed54f9276e018d.

File: tests/unit/test-class-wc-payments-features.php (hunk 2 above)
Comment (brianyu28) [reply tests/unit/test-class-wc-payments-features.php]: Updated in 704ab29f5533d1673e631ad089ed54f9276e018d.

File: tests/unit/test-class-wc-payments-features.php (hunk 4 above)
Comment (brianyu28) [reply tests/unit/test-class-wc-payments-features.php]: Updated in 704ab29f5533d1673e631ad089ed54f9276e018d.

**Comment by brianyu28** [comment]
Thanks @leonardola and @alefesouza! Requesting an additional review from `platform-checkout-reviewers` before merging.

**Review by superdav42 — APPROVED** [review]
LGreatTM!

**Pull request merged**